funnel/views/account.py

```diff
@@ -174,6 +185,106 @@ def siteadmin_comments_spam(self):
 
         return redirect(url_for('siteadmin_comments'))
 
+    @route('siteadmin/review/comments', endpoint='siteadmin_review_comments_random')
+    @requires_login
+    def siteadmin_review_comments_random(self, report=None):
+        if not current_auth.user.is_comment_moderator:
+            return abort(403)
+
+        random_report = CommentModeratorReport.get_one(
+            exclude_reported_by=current_auth.user
+        )
+        if random_report is not None:
+            return redirect(
+                url_for('siteadmin_review_comment', report=random_report.uuid_b58)
+            )
+        else:
+            flash(_("There is no comment report no review at this moment"), 'error')
+            return redirect(url_for('account'))
+
+    @route(
+        'siteadmin/review/comments/<report>',
+        endpoint='siteadmin_review_comment',
+        methods=['GET', 'POST'],
+    )
+    @render_with('siteadmin_review_comment.html.jinja2')
+    @requires_login
+    def siteadmin_review_comment(self, report=None):
+        if not current_auth.user.is_comment_moderator:
+            return abort(403)
+
+        report = CommentModeratorReport.query.filter_by(uuid_b58=report).one_or_404()
+        if report.reported_by == current_auth.user:
+            flash(_("You cannot review same comment twice"), 'error')
```

Comment:
This should amend the existing report.

Or: allow the same comment to be reviewed multiple times, as the moderator may be in error and may need to revise their decision, while the backend should keep earlier reports for archival.

Reply:
We're not keeping existing reports once the fate of the comment is decided. This can amend previous report probably.